lib/mli_parser.ml

```diff
@@ -53,18 +56,68 @@ let slice lines ~(start : Odoc_parser.Loc.point) ~(end_ : Odoc_parser.Loc.point)
       let last_line = String.sub last_line 0 end_.column in
       String.concat "\n" ([ first_line ] @ stripped @ [ last_line ])
 
+let is_newline c = c = '\n'
+
+let find_nth_line s =
+  let max_index = String.length s - 1 in
+  let indexes_of_newlines =
+    s |> String.to_seqi
+    |> Seq.filter_map (fun (i, c) ->
+           match is_newline c with true -> Some i | false -> None)
+  in
+  let indexes_of_line_starts =
+    indexes_of_newlines
+    |> Seq.filter_map (fun i ->
+           match i < max_index with true -> Some (i + 1) | false -> None)
+  in
+  (* first line always starts at index zero, even if there is no preceeding newline *)
+  let indexes = 0 :: List.of_seq indexes_of_line_starts in
+  fun nth ->
+    (* index starts at zero but lines go from 1 *)
```

Comment:
Good job at figuring all these out!

Let's hope `odoc-parser` does not change those facts…